### DISCUSSION TITLE
In tests, what do others think about adding `dbg!`. Idea being provided additional info when tests fail.

### DISCUSSION THREAD

**Discussion by Ben-PH** [massalabs/massa#3487] [Code Style]
e.g. add to a code style guide something like this:

> When writing tests, consider how a test might fail, and add `dbg!` statements where they would be of particular use.  consider the following:
> 
 > - The value of self-documentation. When an error occurs, the `dbg!` should indicate a section of the test that should be considered more carefully in the failure
> - Performance. `dbg!` might be expensive when the cached output becomes very large.
  
  The context I was working in that gave me the idea to raise this discussion can be found in #3846 . Having the dbg! _might_ have made understanding the source of the problems easier to grasp.

**Comment by AurelienFT** [comment]
We don't have a coding style guide for now we only match clippy and rustfmt maybe we could add one and this could be one of the rules but it's a more bigger project.
